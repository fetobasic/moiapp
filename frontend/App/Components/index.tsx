import Pressable from './Pressable';
import Header from './Header';
import HeaderSimple from './HeaderSimple';
import Button from './Button';
import ButtonSimple from './ButtonSimple';
import TextInput from './TextInput';
import TextInputPassword from './TextInputPassword';
import CheckBox from './CheckBox';
import InformationRow from './InformationRow';
import Selector from './Selector';
import Switch from './Switch';
import SwitchRow from './SwitchRow';
import CustomDropdown from './CustomDropdown';
import RadioButtons from './RadioButtons';
import CustomPickerSelect from './CustomPickerSelect';
import SingleRadioButton from './SingleRadioButton';
import SelectModal from './SelectModal';
import ExpandTile from './ExpandTile';
import StyledRangeSlider from './StyledRangeSlider';
import ModalWindow from './ModalWindow';
import DateTimePicker from './DateTimePicker';
import SectionWithTitle from './SectionWithTitle';
import Radar from './Radar';
import ConnectionInfo from './ConnectionInfo';
import Spinner from './Spinner';

// Main
import DrawerContent from './Main/DrawerContent';
import DrawerItem from './Main/DrawerItem';
import DeviceRowInfo from './Main/DeviceRowInfo';
import HomeEmpty from './Main/HomeEmpty';
import HomeYeti from './Main/HomeYeti';
import HomeTile from './Main/HomeTile';
import ConnectInfo from './Main/ConnectInfo';
import HomeIcon from './Main/HomeIcon';
import YetiBatteryLevel from './Yeti/YetiBatteryLevel';
import YetiDeviceSettings from './Yeti/YetiDeviceSettings';
import PortStatusRow from './Main/PortStatusRow';
import YetiOutputPorts from './Yeti/YetiOutputPorts';
import YetiEnergy from './Yeti/YetiEnergy';
import YetiChargeProfile from './Yeti/YetiChargeProfile';
import YetiAccessories from './Yeti/YetiAccessories';
import YetiTemperatureLevel from './Yeti/YetiTemperatureLevel';
import YetiPowerInfo from './Yeti/YetiPowerInfo';
import SimpleSlider from './SimpleSlider';
import Info from './Main/Info';

// Fridge
import HomeFridge from './Fridge/HomeScreen';

// Pairing
import StartInfo from './Pairing/StartInfo';
import UsingWiFi from './Pairing/UsingWiFi';
import ScanedDevices from './Pairing/ScanedDevices';
import InfoRow from './Pairing/InfoRow';
import ProgressLoader from './Pairing/ProgressLoader';
import SelectWiFi from './Pairing/SelectWiFi';

import DisplayInfo from './Settings/DisplayInfo';

// Profile
import CustomProfileInfo from './ChargeProfile/CustomProfileInfo';

// Common
import ValidationErrorList from './ValidationErrorList';

export {
  Pressable,
  Header,
  HeaderSimple,
  Button,
  ButtonSimple,
  Radar,
  DrawerContent,
  DrawerItem,
  CheckBox,
  DeviceRowInfo,
  StartInfo,
  UsingWiFi,
  ScanedDevices,
  InfoRow,
  ProgressLoader,
  SelectWiFi,
  TextInput,
  TextInputPassword,
  HomeEmpty,
  HomeYeti,
  HomeTile,
  ConnectInfo,
  HomeIcon,
  YetiBatteryLevel,
  YetiDeviceSettings,
  PortStatusRow,
  YetiOutputPorts,
  YetiEnergy,
  YetiChargeProfile,
  YetiAccessories,
  YetiTemperatureLevel,
  YetiPowerInfo,
  InformationRow,
  DisplayInfo,
  Info,
  CustomProfileInfo,
  Selector,
  Switch,
  SwitchRow,
  ValidationErrorList,
  CustomDropdown,
  RadioButtons,
  CustomPickerSelect,
  SingleRadioButton,
  SelectModal,
  ExpandTile,
  StyledRangeSlider,
  SimpleSlider,
  ModalWindow,
  DateTimePicker,
  SectionWithTitle,
  HomeFridge,
  ConnectionInfo,
  Spinner,
};
