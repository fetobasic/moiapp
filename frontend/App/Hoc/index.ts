import WithPressable from './WithPressable';
import WithTopBorder from './WithTopBorder';
import FullBorderInfo from './FullBorderInfo';

export { WithPressable, WithTopBorder, FullBorderInfo };
