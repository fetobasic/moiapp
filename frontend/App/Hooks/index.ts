import {
  useAppDispatch,
  useHomeNavigation,
  useAppSelector,
  useMount,
  usePrevious,
  usePairingNavigation,
  useDrawerNavigation,
  useSettingsNavigation,
  useLoginNavigation,
} from './commonHooks';
import useRadar from './useRadar';
import useOpacity from './useOpacity';
import useTimer from './useTimer';
import useBluetooth from './Pairing/useBluetooth';
import useConnectToYeti from './Pairing/useConnectToYeti';
import useLoginAnimation from './useLoginAnimation';
import useMapDrawerData from './useMapDrawerData';
import useEvents from './useEvents';

export {
  useAppDispatch,
  useHomeNavigation,
  useAppSelector,
  useMount,
  usePrevious,
  usePairingNavigation,
  useDrawerNavigation,
  useSettingsNavigation,
  useLoginNavigation,
  useRadar,
  useBluetooth,
  useOpacity,
  useTimer,
  useConnectToYeti,
  useLoginAnimation,
  useMapDrawerData,
  useEvents,
};
