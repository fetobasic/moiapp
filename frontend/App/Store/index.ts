import createStore from 'App/Store/configureStore';

export const store = createStore();
