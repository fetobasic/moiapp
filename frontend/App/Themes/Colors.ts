const colors = {
  white: 'rgba(255, 255, 255, 1)',
  black: 'rgba(0, 0, 0, 1)',
  transparentWhite: (opacity: string) => `rgba(255, 255, 255, ${opacity})`,
  transparentBlack: (opacity: string) => `rgba(0, 0, 0, ${opacity})`,
  overlay: 'rgba(0, 0, 0, 0.5)',
  darkGray: 'rgba(74, 74, 74, 1)',
  lightGray: 'rgba(154, 153, 153, 1)',
  steel: 'rgba(204, 204, 204, 1)',
  border: 'rgba(54, 54, 54, 1)',
  portWarning: 'rgba(255, 205, 29, 1)',
  yellowBorder: 'rgba(250, 253, 79, 1)',
  portWarningDisable: 'rgba(255, 205, 29, 0.5)',
  note: 'rgba(255, 205, 29, 1)',
  middleGray: 'rgba(225, 225, 225, 1)',
  gray: 'rgba(147, 147, 147, 1)',
  dark: 'rgba(18, 18, 18, 1)',
  darkBrown: 'rgba(30, 27, 28, 1)',
  background: 'rgba(27, 27, 27, 1)',
  backgroundModal: 'rgba(46, 46, 46, 1)',
  backgroundHeader: 'rgba(57, 62, 68, 1)',
  coal: 'rgba(74, 74, 74, 1)',
  green: 'rgba(191, 210, 43, 1)',
  escape: 'rgba(5, 209, 5, 1)',
  lightGreen: 'rgba(5, 209, 5, 1)',
  blue: 'rgba(28, 154, 214, 1)',
  red: 'rgba(255, 83, 29, 1)',
  portError: 'rgba(255, 83, 29, 1)',
  portErrorDisable: 'rgba(254, 82, 29, 0.5)',
  link: 'rgba(38, 168, 236, 1)',
  iconGray: 'rgba(128, 128, 128, 1)',
  transparent: 'rgba(0,0,0,0)',
  ricePaper: 'rgba(255,255,255, 0.75)',
  buttonDisabled: 'rgba(255,255,255, 0.38)',
  disabled: 'rgba(133, 133, 133, 1)',
  ember: 'rgba(164, 0, 48, 0.5)',
  separator: 'rgba(142, 141, 147, 1)',
  greenDisable: 'rgba(190, 209, 43, 0.65)',
  grayDisable: 'rgba(147, 147, 147, 0.65)',
  textInputBackground: 'rgba(255,255,255, 0.07)',
  textInputBackgroundDark: 'rgba(150,150,150, 0.07)',
  disabledSegment: 'rgba(255,255,255, 0.12)',
  listBorder: 'rgba(255,255,255, 0.2)',
  dotNotActive: 'rgba(255,255,255, 0.38)',
  mediumEmphasis: 'rgba(255,255,255, 0.6)',
  highEmphasis: 'rgba(255,255,255, 0.87)',
  phoneId: 'rgba(136,136,136, 1)',
  radarDot: 'rgba(50,50,50, 1)',
  // slider doesn't accept rgba
  sliderSelected: '#bdcc2f',
  sliderBlank: '#3a3a3c',
  sliderThumbColor: '#fff',
  sliderThumbBorderColor: '#fff',
  backgroundShade: 'rgba(0, 0, 0, 0.4)',
  dropdownBackground: 'rgba(46, 46, 46, 1)',
  gradient: 'rgba(108, 108, 108, 1)',
  devicesHubRowBackground: 'rgba(39, 39, 39, 1)',
  segmentSelected: 'rgba(99, 99, 102, 1)',
  segmentBackground: 'rgba(42, 43, 44, 1)',
  combiner: 'rgba(0, 240, 255, 1)',
  severity: {
    green: 'rgba(191, 210, 43, 1)',
    yellow: 'rgba(255, 205, 29, 1)',
    red: 'rgba(255, 83, 29, 1)',
    blue: 'rgba(28, 154, 214, 1)',
    gray: 'rgba(255, 255, 255, 0.87)',
  },
  temperature: {
    min: 'rgba(0, 0, 255, 1)',
    max: 'rgba(237, 28, 36, 1)',
  },
  notification: {
    lightYellow: 'rgba(255, 252, 49, 1)',
    lightBlue: 'rgba(133, 249, 254, 1)',
    lightOrange: 'rgba(255, 149, 49, 1)',
    yellow: 'rgba(255, 245, 47, 1)',
  },
  combinerStatus: {
    available: 'rgba(3, 207, 3, 1)',
    syncing: 'rgba(255, 205, 29, 1)',
    active: 'rgba(28, 154, 214, 1)',
  },
};

export default colors;
