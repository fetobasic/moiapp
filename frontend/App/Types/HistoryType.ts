export enum DateTypes {
  PAST_DAY = 'PAST_DAY',
  PAST_WEEK = 'PAST_WEEK',
  PAST_TWO_WEEKS = 'PAST_TWO_WEEKS',
  PAST_MONTH = 'PAST_MONTH',
  PAST_TWO_MONTHS = 'PAST_TWO_MONTHS',
  PAST_HALF_YEAR = 'PAST_HALF_YEAR',
  PAST_YEAR = 'PAST_YEAR',
  PAST_TWO_YEARS = 'PAST_TWO_YEARS',
}

export enum Toggle {
  ON = 'ON',
  OFF = 'OFF',
}

export type DatePeriodType = {
  label: string;
  value: DateTypes;
};

export type ReportUrlType = { [thingName: string]: { [precision: string]: string | null } };

export type ReportDataType = { [thingName: string]: { [precision: string]: { data: any; timestamp: number } | null } };

export type UsageReportUrlResponse = {
  url: string;
};
