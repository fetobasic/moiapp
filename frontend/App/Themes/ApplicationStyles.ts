import { StyleSheet } from 'react-native';
import Metrics from './Metrics';
import Colors from './Colors';
import { isAndroid } from './Platforms';

// This file is for a reusable grouping of Theme items.
// Similar to an XML fragment layout in Android

const ApplicationStyles = StyleSheet.create({
  mainContainer: {
    flex: 1,
    backgroundColor: Colors.background,
  },
  loginContainer: {
    flex: 1,
    backgroundColor: Colors.background,
  },
  container: {
    flex: 1,
    backgroundColor: Colors.background,
    paddingHorizontal: Metrics.baseMargin,
    paddingBottom: isAndroid ? Metrics.baseMargin : 0,
  },
  section: {
    paddingTop: Metrics.marginSection,
    marginHorizontal: Metrics.baseMargin,
  },
  row: {
    flexDirection: 'row',
  },
  flex: {
    flex: 1,
  },
  disabled: {
    opacity: 0.5,
  },
  textGreen: {
    color: Colors.green,
  },
  textRed: {
    color: Colors.red,
  },
  textBlue: {
    color: Colors.blue,
  },
  textNote: {
    color: Colors.note,
  },
  divider: {
    height: 1,
    marginVertical: Metrics.halfMargin,
    backgroundColor: Colors.border,
  },
});

export default ApplicationStyles;
