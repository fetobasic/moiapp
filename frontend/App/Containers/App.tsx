import React from 'react';
import { Provider } from 'react-redux';
import { ReduxNetworkProvider } from 'react-native-offline';
import FlashMessage from 'react-native-flash-message';
import { SafeAreaProvider } from 'react-native-safe-area-context';

import RootContainer from './RootContainer';

import DebugConfig from 'App/Config/DebugConfig';
import { ApplicationStyles } from 'App/Themes';
import { store } from 'App/Store';

/**
 * Provides an entry point into our application. Both index.ios.js and index.android.js
 * call this component first.
 *
 * We create our Redux store here, put it into a provider and then bring in our
 * RootContainer.
 *
 * We separate like this to play nice with React Native's hot reloading.
 */
const App = () => {
  return (
    <Provider store={store}>
      <SafeAreaProvider style={ApplicationStyles.mainContainer}>
        {/*// @ts-ignore as it seems that ReduxNetworkProvider typings have issue with requiring optional props */}
        <ReduxNetworkProvider>
          <RootContainer />
        </ReduxNetworkProvider>
        <FlashMessage position="top" />
      </SafeAreaProvider>
    </Provider>
  );
};

// allow reactotron overlay for fast design in dev mode
// @ts-ignore as it seems that `console.tron` typings have issue with requiring only React.ReactNode
export default DebugConfig.useReactotron ? console?.tron?.overlay?.(App) : App;
