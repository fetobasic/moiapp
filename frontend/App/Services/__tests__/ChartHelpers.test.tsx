import { render } from '@testing-library/react-native';
import {
  formatLongNumber,
  formatToKilo,
  getDateFormat,
  getFormattedPeakDate,
  getMaxChartY,
  getYFromPercentage,
  renderBottomDates,
  renderTransparentLine,
  tickFormat,
} from 'App/Services/ChartHelpers';
import { Colors } from 'App/Themes';
import { DateTypes } from 'App/Types/HistoryType';

describe('ChartHelpers', () => {
  describe('getMaxChartY', () => {
    const chartData = {
      showEnergyOut: false,
      showEnergyIn: false,
      showBattery: false,
      batterySoc: [],
      whIn: [],
      whOut: [],
    };

    test('should return 100 when no data is shown', () => {
      const result = getMaxChartY(chartData);
      expect(result).toBe(100);
    });

    test('should return the maximum value of batterySoc when only battery data is shown', () => {
      const batterySoc = [{ y: 50 }, { y: 60 }, { y: 70 }];
      const result = getMaxChartY({ ...chartData, showBattery: true, batterySoc });
      expect(result).toBe(100);
    });

    test('should return the maximum value of whIn when only energy in data is shown', () => {
      const whIn = [{ y: 100 }, { y: 150 }, { y: 200 }];
      const result = getMaxChartY({ ...chartData, showEnergyIn: true, whIn });
      expect(result).toBe(300);
    });

    test('should return the maximum value of whOut when only energy out data is shown', () => {
      const whOut = [{ y: 30 }, { y: 40 }, { y: 50 }];
      const result = getMaxChartY({ ...chartData, showEnergyOut: true, whOut });
      expect(result).toBe(150);
    });

    test('should return the sum of max values when multiple data types are shown', () => {
      const batterySoc = [{ y: 70 }, { y: 80 }, { y: 90 }];
      const whIn = [{ y: 150 }, { y: 200 }, { y: 250 }];
      const whOut = [{ y: 40 }, { y: 50 }, { y: 60 }];
      const result = getMaxChartY({
        showEnergyOut: true,
        showEnergyIn: true,
        showBattery: true,
        batterySoc,
        whIn,
        whOut,
      });
      expect(result).toBe(250 + 60); // Max of whIn + Max of whOut
    });
  });

  describe('tickFormat', () => {
    test('should return an empty string', () => {
      const result = tickFormat('', 0, DateTypes.PAST_MONTH);

      expect(result).toBe('');
    });

    test('should return the formatted date for date type', () => {
      const evenIndexResult = tickFormat('2023-5-16:21-40', 0, DateTypes.PAST_TWO_YEARS);

      expect(evenIndexResult).toBe('May');
    });

    test('should return an empty string for past day', () => {
      const result = tickFormat('2023-10-16:16-20', 1, DateTypes.PAST_DAY);

      expect(result).toBe('');
    });

    test('should return formatted date with non even index', () => {
      const result = tickFormat('2023-10-16:16-20', 3, DateTypes.PAST_TWO_WEEKS);

      expect(result).toBe('M');
    });
  });

  describe('renderTransparentLine', () => {
    const whIn = [{ x: 1 }, { x: 2 }, { x: 3 }];
    const maxY = 100;
    const result = renderTransparentLine({ whIn, maxY });

    test('should generate the correct data for VictoryLine', () => {
      const { data } = result.props;

      expect(data).toEqual(whIn.map(({ x }) => ({ x, y: maxY })));
    });

    test('should set the correct style properties', () => {
      const { style } = result.props;

      expect(style).toEqual({ data: { stroke: Colors.transparent, strokeWidth: 0 } });
    });
  });

  test('should calculate the correct Y value from a percentage', () => {
    const testCases = [
      { percentage: 0, maxY: 100, expectedY: 0 },
      { percentage: 50, maxY: 200, expectedY: 100 },
      { percentage: 75, maxY: 300, expectedY: 225 },
      { percentage: 100, maxY: 50, expectedY: 50 },
    ];

    testCases.forEach(({ percentage, maxY, expectedY }) => {
      const result = getYFromPercentage({ percentage, maxY });

      expect(result).toBe(expectedY);
    });
  });

  describe('formatToKilo', () => {
    test('should format numbers less than 1000 as-is', () => {
      const testCases = [
        { num: 0, expectedFormatted: '0 ' },
        { num: 999, expectedFormatted: '999 ' },
        { num: -500, expectedFormatted: '-500 ' },
      ];

      testCases.forEach((testCase) => {
        const { num, expectedFormatted } = testCase;
        const result = formatToKilo(num);
        expect(result).toBe(expectedFormatted);
      });
    });

    test('should format numbers greater than or equal to 1000000 without "K" suffix', () => {
      const result = formatToKilo(100000);
      expect(result).toBe('100 K');
    });

    test('should format numbers greater than or equal to 1000000 with "K" suffix', () => {
      const result = formatToKilo(1000000);
      expect(result).toBe('1000 K');
    });

    test('should use the provided suffix', () => {
      const result = formatToKilo(7000000, 'M');
      expect(result).toBe('7000 M');
    });
  });

  describe('formatLongNumber', () => {
    test('should format numbers less than 1,000,000 using formatToKilo', () => {
      const testCases = [
        { num: 0, expectedFormatted: '0 ' },
        { num: 999, expectedFormatted: '999 ' },
        { num: 50000, expectedFormatted: '50 K' },
        { num: 999999, expectedFormatted: '999 K' },
      ];

      testCases.forEach((testCase) => {
        const { num, expectedFormatted } = testCase;
        const result = formatLongNumber(num);
        expect(result).toBe(expectedFormatted);
      });
    });

    test('should format numbers greater than or equal to 1,000,000 as millions with a comma separator', () => {
      const testCases = [
        { num: 1000000, expectedFormatted: '1 M' },
        { num: 9000000, expectedFormatted: '9 M' },
      ];

      testCases.forEach((testCase) => {
        const { num, expectedFormatted } = testCase;
        const result = formatLongNumber(num);
        expect(result).toBe(expectedFormatted);
      });
    });
  });

  describe('getDateFormat', () => {
    test('should return the correct date format for DateTypes.PAST_DAY', () => {
      const result = getDateFormat(DateTypes.PAST_DAY);
      expect(result).toBe('h');
    });

    test('should return the correct date format for DateTypes.PAST_TWO_WEEKS', () => {
      const result = getDateFormat(DateTypes.PAST_TWO_WEEKS);
      expect(result).toBe('EEEEE');
    });

    test('should return the correct date format for DateTypes.PAST_MONTH', () => {
      const result = getDateFormat(DateTypes.PAST_MONTH);
      expect(result).toBe('d');
    });

    test('should return the correct date format for DateTypes.PAST_YEAR', () => {
      const result = getDateFormat(DateTypes.PAST_YEAR);
      expect(result).toBe('LLL');
    });

    test('should return the correct date format for DateTypes.PAST_TWO_YEARS', () => {
      const result = getDateFormat(DateTypes.PAST_TWO_YEARS);
      expect(result).toBe('LLL');
    });
  });

  describe('renderBottomDates', () => {
    const maxY = 100;

    test('should render VictoryAxis elements for DateTypes', () => {
      Object.values(DateTypes).forEach((tempUsage) => {
        const whIn = [{ x: '2023-10-16:10-00' }, { x: '2023-10-12:00-00' }];
        const screen = render(<div>{renderBottomDates({ maxY, whIn, tempUsage })}</div>).toJSON();
        expect(screen).toMatchSnapshot();
      });
    });

    test('should return null for an unknown DateTypes value', () => {
      const tempUsage = 'UnknownType';
      const whIn = [];
      const result = renderBottomDates({ maxY, whIn, tempUsage });
      expect(result).toBeNull();
    });
  });

  describe('getFormattedPeakDate', () => {
    const tempUsage = [
      { date: DateTypes.PAST_DAY, inputDate: '2023-10-16 15:30:00', expectDate: '3PM, Oct 16 2023' },
      { date: DateTypes.PAST_TWO_WEEKS, inputDate: '2023-08-02 18:57:03', expectDate: '6PM, Aug 02 2023' },
      { date: DateTypes.PAST_MONTH, inputDate: '2023-05-03 07:47:20', expectDate: 'May 03, 2023' },
      { date: DateTypes.PAST_YEAR, inputDate: '2022-11-12 12:20:10', expectDate: 'Nov 12, 2022' },
      { date: DateTypes.PAST_TWO_YEARS, inputDate: '2021-01-15 17:30:00', expectDate: 'Jan 15, 2021' },
    ];
    tempUsage.forEach(({ date, inputDate, expectDate }) => {
      test(`should format date for ${date}`, () => {
        const formattedDate = getFormattedPeakDate(date, inputDate);
        expect(formattedDate).toBe(expectDate);
      });
    });
  });
});
