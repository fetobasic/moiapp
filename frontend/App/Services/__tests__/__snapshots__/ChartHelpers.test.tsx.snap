// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartHelpers renderBottomDates should render VictoryAxis elements for DateTypes 1`] = `
<div>
  <View
    axisLabelComponent={
      <UNDEFINED
        angle={0}
        dx={15}
        dy={152}
      />
    }
    axisValue={0}
    dependentAxis={true}
    domain={
      [
        -14,
        0,
      ]
    }
    label="AM"
    style={
      {
        "axis": {
          "stroke": "none",
        },
        "axisLabel": {
          "fill": "rgba(255, 255, 255, 0.87)",
          "fontSize": 10,
        },
        "grid": {
          "stroke": "none",
        },
        "tickLabels": {
          "fill": "none",
        },
        "ticks": {
          "stroke": "none",
        },
      }
    }
  />
  <View
    axisLabelComponent={
      <UNDEFINED
        angle={0}
        dx={28}
        dy={152}
      />
    }
    axisValue={1}
    dependentAxis={true}
    domain={
      [
        -14,
        0,
      ]
    }
    label=""
    style={
      {
        "axis": {
          "stroke": "none",
        },
        "axisLabel": {
          "fill": "rgba(255, 255, 255, 0.87)",
          "fontSize": 10,
        },
        "grid": {
          "stroke": "none",
        },
        "tickLabels": {
          "fill": "none",
        },
        "ticks": {
          "stroke": "none",
        },
      }
    }
  />
</div>
`;

exports[`ChartHelpers renderBottomDates should render VictoryAxis elements for DateTypes 2`] = `<div />`;

exports[`ChartHelpers renderBottomDates should render VictoryAxis elements for DateTypes 3`] = `
<div>
  <View
    axisLabelComponent={
      <UNDEFINED
        angle={0}
        dx={28}
        dy={152}
      />
    }
    axisValue={0.5}
    dependentAxis={true}
    domain={
      [
        -14,
        0,
      ]
    }
    label="Oct 16"
    style={
      {
        "axis": {
          "stroke": "rgba(54, 54, 54, 1)",
        },
        "axisLabel": {
          "fill": "rgba(255, 255, 255, 0.87)",
          "fontSize": 10,
        },
        "grid": {
          "stroke": "none",
        },
        "tickLabels": {
          "fill": "none",
        },
        "ticks": {
          "stroke": "none",
        },
      }
    }
  />
  <View
    axisLabelComponent={
      <UNDEFINED
        angle={0}
        dx={28}
        dy={152}
      />
    }
    axisValue={1.5}
    dependentAxis={true}
    domain={
      [
        -14,
        0,
      ]
    }
    label=""
    style={
      {
        "axis": {
          "stroke": "none",
        },
        "axisLabel": {
          "fill": "rgba(255, 255, 255, 0.87)",
          "fontSize": 10,
        },
        "grid": {
          "stroke": "none",
        },
        "tickLabels": {
          "fill": "none",
        },
        "ticks": {
          "stroke": "none",
        },
      }
    }
  />
</div>
`;

exports[`ChartHelpers renderBottomDates should render VictoryAxis elements for DateTypes 4`] = `
<div>
  <View
    axisLabelComponent={
      <UNDEFINED
        angle={0}
        dx={15}
        dy={152}
      />
    }
    axisValue={0}
    dependentAxis={true}
    domain={
      [
        -14,
        0,
      ]
    }
    label="Oct"
    style={
      {
        "axis": {
          "stroke": "none",
        },
        "axisLabel": {
          "fill": "rgba(255, 255, 255, 0.87)",
          "fontSize": 10,
        },
        "grid": {
          "stroke": "none",
        },
        "tickLabels": {
          "fill": "none",
        },
        "ticks": {
          "stroke": "none",
        },
      }
    }
  />
  <View
    axisLabelComponent={
      <UNDEFINED
        angle={0}
        dx={28}
        dy={152}
      />
    }
    axisValue={1}
    dependentAxis={true}
    domain={
      [
        -14,
        0,
      ]
    }
    label=""
    style={
      {
        "axis": {
          "stroke": "none",
        },
        "axisLabel": {
          "fill": "rgba(255, 255, 255, 0.87)",
          "fontSize": 10,
        },
        "grid": {
          "stroke": "none",
        },
        "tickLabels": {
          "fill": "none",
        },
        "ticks": {
          "stroke": "none",
        },
      }
    }
  />
</div>
`;

exports[`ChartHelpers renderBottomDates should render VictoryAxis elements for DateTypes 5`] = `<div />`;

exports[`ChartHelpers renderBottomDates should render VictoryAxis elements for DateTypes 6`] = `<div />`;

exports[`ChartHelpers renderBottomDates should render VictoryAxis elements for DateTypes 7`] = `
<div>
  <View
    axisLabelComponent={
      <UNDEFINED
        angle={0}
        dx={15}
        dy={152}
      />
    }
    axisValue={0}
    dependentAxis={true}
    domain={
      [
        -14,
        0,
      ]
    }
    label="2023"
    style={
      {
        "axis": {
          "stroke": "none",
        },
        "axisLabel": {
          "fill": "rgba(255, 255, 255, 0.87)",
          "fontSize": 10,
        },
        "grid": {
          "stroke": "none",
        },
        "tickLabels": {
          "fill": "none",
        },
        "ticks": {
          "stroke": "none",
        },
      }
    }
  />
  <View
    axisLabelComponent={
      <UNDEFINED
        angle={0}
        dx={28}
        dy={152}
      />
    }
    axisValue={1.5}
    dependentAxis={true}
    domain={
      [
        -14,
        0,
      ]
    }
    label=""
    style={
      {
        "axis": {
          "stroke": "none",
        },
        "axisLabel": {
          "fill": "rgba(255, 255, 255, 0.87)",
          "fontSize": 10,
        },
        "grid": {
          "stroke": "none",
        },
        "tickLabels": {
          "fill": "none",
        },
        "ticks": {
          "stroke": "none",
        },
      }
    }
  />
</div>
`;

exports[`ChartHelpers renderBottomDates should render VictoryAxis elements for DateTypes 8`] = `
<div>
  <View
    axisLabelComponent={
      <UNDEFINED
        angle={0}
        dx={15}
        dy={152}
      />
    }
    axisValue={0}
    dependentAxis={true}
    domain={
      [
        -14,
        0,
      ]
    }
    label="2023"
    style={
      {
        "axis": {
          "stroke": "none",
        },
        "axisLabel": {
          "fill": "rgba(255, 255, 255, 0.87)",
          "fontSize": 10,
        },
        "grid": {
          "stroke": "none",
        },
        "tickLabels": {
          "fill": "none",
        },
        "ticks": {
          "stroke": "none",
        },
      }
    }
  />
  <View
    axisLabelComponent={
      <UNDEFINED
        angle={0}
        dx={28}
        dy={152}
      />
    }
    axisValue={1}
    dependentAxis={true}
    domain={
      [
        -14,
        0,
      ]
    }
    label=""
    style={
      {
        "axis": {
          "stroke": "none",
        },
        "axisLabel": {
          "fill": "rgba(255, 255, 255, 0.87)",
          "fontSize": 10,
        },
        "grid": {
          "stroke": "none",
        },
        "tickLabels": {
          "fill": "none",
        },
        "ticks": {
          "stroke": "none",
        },
      }
    }
  />
</div>
`;
